include/nbl/builtin/hlsl/shapes/triangle.hlsl

```diff
@@ -0,0 +1,116 @@
+
+#ifndef _NBL_BUILTIN_HLSL_SHAPES_TRIANGLE_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SHAPES_TRIANGLE_INCLUDED_
+
+#include <nbl/builtin/hlsl/common.hlsl>
+#include <nbl/builtin/hlsl/math/functions.hlsl>
+#include <nbl/builtin/hlsl/limits/numeric.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+
+namespace shapes
+{
+
+//
+float3x3 getSphericalTriangle(in float3x3 vertices, in float3 origin)
+{
+	float3x3 tp_vertices = transpose(vertices);
```

Comment:
Look here

https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/builtin/glsl/shapes/triangle.glsl#L7-L11

the `tranposes` cancel out, so it actually breaks things